shimmingtoolbox/cli/mask.py

```diff
@@ -128,3 +130,153 @@ def threshold(fname_input, output, thr):
+              help="Choice of kernel shape for the CNN. Segmentation with 3D kernels is slower than with "
+                   "2D kernels.")
+@click.option('--remove-tmp', 'remove_tmp', type=bool, default=True, show_default=True,
+              help="Remove temporary files.")
+@click.option('--verbose', type=click.IntRange(0, 2), default=1, show_default=True,
+              help="Verbose: 0 = nothing, 1 = classic, 2 = expended.")
+# Options for _get_centerline
+# @click.option('--method', type=click.Choice(['optic', 'fitseg']), default='optic',
+#               help="(str): Method used for extracting the centerline: "
+#                    "- optic: automatic spinal cord detection method"
+#                    "- fitseg: fit a regularized centerline on an already-existing cord segmentation. It will "
+#                    "interpolate if slices are missing and extrapolate beyond the segmentation boundaries (i.e., "
+#                    "every axial slice will exhibit a centerline pixel). (default: optic)")
+# @click.option('--centerline_algo', type=click.Choice(['polyfit', 'bspline', 'linear', 'nurbs']), default='bspline',
+#               help="(str): Algorithm for centerline fitting. Only relevant with -method fitseg (default: bspline)")
+# @click.option('--centerline_smooth', default=30, help="(int): Degree of smoothing for centerline fitting. Only for "
+#                                                      "-centerline-algo {bspline, linear}. (default: 30)")
+def sct(fname_input, fname_output, contrast, centerline, file_centerline, brain, kernel, size, shape, remove_tmp,
+        verbose):
+
+    # Make sure input path exists
+    if not os.path.exists(fname_input):
+        raise RuntimeError("Input file does not exist")
+
+    # Get the number of dimensions
+    nii_input = nib.load(fname_input)
+    ndim = nii_input.ndim
+    # If 4d, last dimension is time, average last dim for better SNT
+    if ndim == 4:
+        input_3d = np.mean(nii_input.get_fdata(), 3)
+        nii_3d = nib.Nifti1Image(input_3d, affine=nii_input.affine, header=nii_input.header)
+        fname_mean = os.path.join(os.path.dirname(fname_output), 'mean_3d.nii.gz')
+        nib.save(nii_3d, fname_mean)
+        fname_process = fname_mean
+    # If not then only set the processing filename
+    else:
+        fname_process = fname_input
+
+    fname_seg = os.path.join(os.path.dirname(fname_output), 'seg.nii.gz')
+
+    # sct_get_centerline is faster than sct_deepseg_sc, however, it is a bit less accurate. More investigations needed
+    # in the future, this code is commented out so that we can persue investigation.
+    # # Get the centerline
+    # _get_centerline(fname_process, fname_seg)
+
+    # Run sct_deepseg_sc
+    # Use sct parameter convention
+    if remove_tmp:
+        remove = 1
+    else:
+        remove = 0
+
+    cmd = f"sct_deepseg_sc -i {fname_process} -o {fname_seg} -c {contrast} -centerline {centerline} -kernel {kernel} " \
```

Comment:
sct_deepseg_sc does not have -o as an option. We can use -ofolder instead, in which case it seems that the centerline segmentation file will be named the same as the input file with _seg appended at the end.

Reply:
I do have the option. Here is the output of `sct_deepseg_sc -h`

<details>
  <summary>Terminal output</summary>
  
```bash
sct_deepseg_sc -h
--

usage: sct_deepseg_sc -i <file> -c {t1,t2,t2s,dwi} [-h] [-centerline {svm,cnn,viewer,file}] [-file_centerline <str>] [-thr <float>] [-brain {0,1}] [-kernel {2d,3d}] [-ofolder <str>] [-o <file>] [-r {0,1}] [-v <int>] [-qc <str>]
                      [-qc-dataset <str>] [-qc-subject <str>] [-igt <str>]

Spinal Cord Segmentation using convolutional networks. Reference: Gros et al. Automatic segmentation of the spinal cord and intramedullary multiple sclerosis lesions with convolutional neural networks. Neuroimage. 2019 Jan 1;184:901-915.

MANDATORY ARGUMENTS:
  -i <file>             Input image. Example: t1.nii.gz
  -c {t1,t2,t2s,dwi}    Type of image contrast.

OPTIONAL ARGUMENTS:
  -h, --help            show this help message and exit
  -centerline {svm,cnn,viewer,file}
                        Method used for extracting the centerline:
                         svm: Automatic detection using Support Vector Machine algorithm.
                         cnn: Automatic detection using Convolutional Neural Network.
                         viewer: Semi-automatic detection using manual selection of a few points with an interactive viewer followed by regularization.
                         file: Use an existing centerline (use with flag -file_centerline) (default: svm)
  -file_centerline <str>
                        Input centerline file (to use with flag -centerline file). Example: t2_centerline_manual.nii.gz
  -thr <float>          Binarization threshold (between 0 and 1) to apply to the segmentation prediction. Set to -1 for no binarization (i.e. soft segmentation output). The default threshold is specific to each contrast and
                        wasestimated using an optimization algorithm. More details at: https://github.com/sct-pipeline/deepseg-threshold.
  -brain {0,1}          Indicate if the input image contains brain sections (to speed up segmentation). Only use with "-centerline cnn".
  -kernel {2d,3d}       Choice of kernel shape for the CNN. Segmentation with 3D kernels is slower than with 2D kernels. (default: 2d)
  -ofolder <str>        Output folder. Example: My_Output_Folder/ (default: /Users/alex/Documents/School/Polytechnique/Master/project/shimming-toolbox)
  -o <file>             Output filename. Example: spinal_seg.nii.gz
  -r {0,1}              Remove temporary files. (default: 1)
  -v <int>              Verbosity. 0: Display only errors/warnings, 1: Errors/warnings + info messages, 2: Debug mode (default: 1)
  -qc <str>             The path where the quality control generated content will be saved
  -qc-dataset <str>     If provided, this string will be mentioned in the QC report as the dataset the process was run on
  -qc-subject <str>     If provided, this string will be mentioned in the QC report as the subject the process was run on
  -igt <str>            File name of ground-truth segmentation.
```
</details>


You can see the -o option. Maybe update sct?

Reply:
I Updated SCT and now have the -o option! No issues now.